workshop/content/lab1.5_kafka.md

```diff
@@ -0,0 +1,272 @@
+```
+
+Verify that it was created.
+
+```
+oc get kafkasource
+```
+
+```
+NAME               TOPICS                     BOOTSTRAPSERVERS                        READY   REASON   AGE
+mykafka-source     my-topic-[user-number]     my-cluster-kafka-bootstrap.kafka:9092   True             76s
+```
+
+Using the web browser, click on over to the `Developer` view.  Then click `Topology` and make sure you are in the correct `Project`.  You can then verify the KafkaSource is configured correctly.
+
+![Serverless Kafka Source](./images/serverless_kafka_source.png)
+
+### See It In Action
+Now let's see events flow through the Serverless system. Create the kafka producer again.
+
+In one terminal run:
+
+```
+oc run kafka-producer -ti --image=strimzi/kafka:0.19.0-kafka-2.5.0 --rm=true --restart=Never -- bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap.kafka:9092 --topic my-topic-$USER_NUMBER
+```
+
+In another terminal run
+
+```
+stern eventinghello -c user-container
+```
+
+Every message you type into the producer, will appear in the knative service logs (that you are viewing through `stern`), along with other CloudEvent metadata.
+
+Let's make sure the pods were properly deleted.
+
+```
+oc delete kafka-producer
+# it's ok to get "error: the server doesn't have a resource type "kafka-producer""
+```
+
+### Turn on the Firehose
+Time to add more events and watch the system process them.
+
+In one terminal watch the pods
+
+```
+watch oc get pods
+```
+
+In another terminal, run the kafka-spammer.
+
+```
+oc run kafka-spammer -it --image=jonnyman9/kafka-python-spammer:latest --rm=true --restart=Never --env KAFKA_BOOTSTRAP_HOST=my-cluster-kafka-bootstrap.kafka --env TOPIC_NAME=my-topic-$USER_NUMBER --env TIMES=10
+```
+
+In the first terminal watching the pods, you should see 1 or more new pods.
+
+```
+NAME                                                              READY   STATUS    RESTARTS   AGE
+eventinghello-v1-deployment-f48945f8b-56sal                       2/2     Running   0          9s
+```
+
+If you want to, feel free to open another terminal to watch the logs
+
+```
+stern eventinghello -c user-container
+```
+
+And try to run the `kafka-spammer` again, this time, increasing `TIMES` to 100 or more.
+
+```
+oc delete pod kafka-spammer
+oc run kafka-spammer -it --image=jonnyman9/kafka-python-spammer:latest --rm=true --restart=Never --env KAFKA_BOOTSTRAP_HOST=my-cluster-kafka-bootstrap.kafka --env TOPIC_NAME=my-topic-$USER_NUMBER --env TIMES=100
+```
+
+The Serverless system will spin up enough Serverless applications that are required in order to keep up with processing of the Kafka events.  Our application doesn't do much and as a result is able to keep up the volume of events that we push through the system.
+
+###  Cleanup
+
+```
+curl -sS https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/main/kafka/kafka-source-to-sink.yml | sed "s/USER_NUMBER/$USER_NUMBER/" | oc delete -f -
```

Comment:
I would write something like the following instead:

`oc delete kafkatopic my-topic-$USER_NUMBER'

Reply:
👍